### ISSUE TITLE
GUI Frontend for Sirji

### ISSUE THREAD

**Issue by poornima-alugubelly** [sirji-ai/sirji#7]
Have been toying around building something similar.

Continuing the efforts, I have tweaked and built the frontend for Sirji - https://sirji-ai-fe-demo-duiq.vercel.app/

It is not fully functional yet, will wait for Sirji to be full developed before making it completely useable. 

Do have a look, and let me know what do you think. Would love to contribute further.

Repo link:  https://github.com/poornima-alugubelly/sirji-ai-fe-demo 
Tech stack used:
- Nextjs 14 
- Tailwind CSS
- Zustand

Todo:
Integrate `workspace`:
- logger
- Browser
- shell
- planner

![image](https://github.com/sirji-ai/sirji/assets/70846516/1837ec83-e172-463c-bffe-0c3c27ff2382)

**Comment by sunilkhedar** [comment]
@poornima-alugubelly Can you please complete the integration for browser, shell, planner and editor?

**Comment by poornima-alugubelly** [comment]
@sunilkhedar Hi yes, I'll try to finish it soon

**Comment by poornima-alugubelly** [comment]
@sunilkhedar Can you please point me towards the apis for using browser, shell, planner and editor? I went through the code but was not able to figure out how exactly we are exposing these functionalities

**Comment by sunilkhedar** [comment]
@poornima-alugubelly The current release is focused on VS Code Extension. It won't need any APIs. 

Can you propose the APIs and the presentation layer? 

We have published the agents, tools, and messages Python packages for reusability. Here are the packages: https://pypi.org/search/?q=sirji

**Comment by poornima-alugubelly** [comment]
> @poornima-alugubelly The current release is focused on VS Code Extension. It won't need any APIs.
> 
> Can you propose the APIs and the presentation layer?
> 
> We have published the agents, tools, and messages Python packages for reusability. Here are the packages: https://pypi.org/search/?q=sirji

Sure I will look into this

**Comment by poornima-alugubelly** [comment]
@sunilkhedar I have tried to integrate with FE and was trying to use WebSockets as they would be the most appropriate for this use case, however I faced major roadblocks while modifying the BE logic and I am not really familiar with BE development and Python. As a result, I won’t be able to solve it effectively at this time. If someone else with more expertise is available, they might be better suited to tackle this.

Thank you for acknowledging my contributions!
